src/github.com/couchbase/sync_gateway/base/bucket.go

```diff
@@ -121,10 +148,212 @@ func (bucket CouchbaseBucket) StartTapFeed(args sgbucket.TapArguments) (sgbucket
 		return bucket.StartCouchbaseTapFeed(args)
+
+	}
+
+}
+
+func (bucket CouchbaseBucket) initCBGTManager() error {
+
+	var username, password string
+
+	feedParams := bucket.spec.FeedParams
+
+	if bucket.spec.Auth != nil {
+		username, password, _ = bucket.spec.Auth.GetCredentials()
+	}
+
+	if username == "" {
+		username = bucket.Name
 	}
+
+	couchbaseUrl, err := CouchbaseUrlWithAuth(
+		bucket.spec.Server,
+		username,
+		password,
+		bucket.Name,
+	)
+	if err != nil {
+		return err
+	}
+
+	// the connection string, eg: "couchbase:http://user:pass@localhost:8091"
+	connect := fmt.Sprintf("couchbase:%v", couchbaseUrl)
+
+	uuid, err := cmd.MainUUID(IndexTypeSyncGateway, feedParams.DataDir)
+	if err != nil {
+		return err
+	}
+
+	// this tells CBGT that we are brining a new CBGT node online
+	register := "wanted"
+
+	// use the uuid as the bindHttp so that we don't have to make the user
+	// configure this, and since as far as the REST Api interface, we'll be using
+	// whatever is configured in adminInterface anyway.
+	// More info here:
+	//   https://github.com/couchbaselabs/cbgt/issues/1
+	//   https://github.com/couchbaselabs/cbgt/issues/25
+	bindHttp := uuid
+
+	cfg, err := cmd.MainCfg(
+		IndexTypeSyncGateway,
+		connect,
+		bindHttp,
+		register,
+		feedParams.DataDir,
+	)
+	if err != nil {
+		return err
+	}
+
+	// type assertion to convert to CfgCB so we can call SetKeyPrefix
+	cfgCb, ok := cfg.(*cbgt.CfgCB)
+	if !ok {
+		return fmt.Errorf("Type assertion failure")
+	}
+
+	// this will cause CBGT to name the cfg document _sync:cfg
+	cfgCb.SetKeyPrefix("_sync:")
+
+	tags := []string{"feed", "janitor", "pindex", "planner"}
+	container := ""
+	weight := 1                  // this would allow us to have more pindexes serviced by this node
+	server := bucket.spec.Server // or use "." (don't bother checking)
+	extras := ""
+	var managerEventHandlers cbgt.ManagerEventHandlers
+
+	// refresh it so we have a fresh copy
+	if err := cfg.Refresh(); err != nil {
+		return err
+	}
+
+	CBGTManager = cbgt.NewManager(
+		cbgt.VERSION,
+		cfgCb,
+		uuid,
+		tags,
+		container,
+		weight,
+		extras,
+		bindHttp,
+		feedParams.DataDir,
+		server,
+		managerEventHandlers,
+	)
+	err = CBGTManager.Start(register)
+	if err != nil {
+		log.Printf("Manager.Start() returned error: %v", err)
+		return err
+	}
+
+	return nil
+
+}
+
+func (bucket CouchbaseBucket) checkCBGTIndexExists(indexName string) (bool, error) {
+
+	_, indexDefsMap, err := CBGTManager.GetIndexDefs(true)
+	if err != nil {
+		return false, err
+	}
+
+	return (indexDefsMap[indexName] != nil), nil
+
+}
+
+// Create an "index" in CBGT which will cause it to start streaming
+// DCP events to us for our shard of the full DCP stream.
+func (bucket CouchbaseBucket) createCBGTIndex() error {
+
+	authHandler := bucket.getDcpAuthHandler()
+	user, pwd, _ := authHandler.GetCredentials()
+	if user == "" && bucket.Name != "default" {
+		user = bucket.Name
+	}
```

Comment:
dcpAuth.GetCredentials() is already doing some of this, but not the 'default' check.  If the name=default check is standard, we should move it into dcpAuth.GetCredentials().